sdk/dotNet/Integrations/AzureKeyVault/AzureKeyValueStorage.cs

```
@@ -0,0 +1,384 @@
+        {
+            if (config.Count == 0)
+            {
+                LoadConfigAsync().Wait();
+            }
+            return config.TryGetValue(key, out var value) ? value : null;
+        }
+
+        public void SaveString(string key, string value)
+        {
+            if (config.Count == 0)
+            {
+                LoadConfigAsync().Wait();
+            }
+            config[key] = value;
+            SaveConfigAsync(config).Wait();
+        }
+
+        public byte[]? GetBytes(string key)
+        {
+            if (config.Count == 0)
+            {
+                LoadConfigAsync().Wait();
+            }
+
+            var stringValue = config.TryGetValue(key, out var result) ? result : null;
+            return stringValue == null ? null : CryptoUtils.Base64ToBytes(stringValue);
+        }
+
+        public void SaveBytes(string key, byte[] value)
+        {
+            if (config.Count == 0)
+            {
+                LoadConfigAsync().Wait();
+            }
+            config[key] = CryptoUtils.BytesToBase64(value);
+            SaveConfigAsync(config).Wait();
+        }
+
+        public void Delete(string key)
+        {
+            config.Remove(key);
+            SaveConfigAsync(config).Wait();
+        }
+
+        public async Task CreateConfigFileIfMissingAsync()
+        {
+            try
+            {
+                if (File.Exists(configFileLocation))
+                {
+                    logger.LogInformation("Config file already exists at: {Path}", configFileLocation);
+                    return;
+                }
+
+                logger.LogInformation("Config file does not exist at: {Path}", configFileLocation);
+                string? directory = Path.GetDirectoryName(configFileLocation);
+                if (directory != null && !Directory.Exists(directory))
+                {
+                    Directory.CreateDirectory(directory);
+                }
+
```

you can add else with throw for an error in directory creation or invalid directory path
  else
        {
            throw new InvalidOperationException("");
        }

---

By default, CreateDicrectory is throwing these exceptions:
Exceptions:
       T:System.ArgumentException:
         path is a zero-length string, or contains one or more invalid characters. You
         can query for invalid characters by using the System.IO.Path.GetInvalidPathChars
         method.
        //
       T:System.ArgumentException:
         The file mode is invalid.
        //
       T:System.ArgumentNullException:
         path is null.
        //
       T:System.UnauthorizedAccessException:
         The caller does not have the required permission.
        //
       T:System.IO.PathTooLongException:
         The specified path exceeds the system-defined maximum length.
        //
       T:System.IO.IOException:
         path is a file.
        //
       T:System.IO.DirectoryNotFoundException:
         A component of the path is not a directory.
         
  These look pretty self explanatory, do you think maybe logging the exception and rethrowing will add more context here than just catching and processing the error ?